src/node.rs

```diff
@@ -321,6 +321,87 @@ impl PbftNode {
         }
     }
 
+    /// Verifies the consensus seal from the current block, for the previous block
+    fn verify_consensus_seal(&mut self, block: &Block) -> Result<(), PbftError> {
+        // We don't publish a consensus seal until block 2, so we don't verify it
+        // until block 3
+        if block.block_num < 3 {
+            return Ok(());
+        }
+
+        if block.payload.is_empty() {
+            return Err(PbftError::InternalError(
+                "Got empty payload for non-genesis block!".into(),
+            ));
+        }
+
+        let seal: PbftSeal =
+            protobuf::parse_from_bytes(&block.payload).map_err(PbftError::SerializationError)?;
+
+        let id_matches = seal.previous_id == &block.previous_id[..];
+        let summary_matches = seal.summary == &block.summary[..];
+
+        if !(id_matches && summary_matches) {
+            warn!(
+                "Consensus seal failed verification. ID matched? {} Summary matched? {}",
+                id_matches, summary_matches
+            );
+
+            return Err(PbftError::InternalError(
+                "Consensus seal verification error!".into(),
+            ));
+        }
+
+        let context = create_context("secp256k1").unwrap();
+
+        for vote in seal.get_previous_commit_votes() {
```

Comment:
Two things that aren't being checked here but should be:
1. We are not verifying the number of votes (we need to make sure there are 2f+1)
2. We are not verifying that all of the votes are from nodes in the network (we need to check this against the list of node ids stored in the node's state)